### PR TITLE
Fix Vevo titles with featured artists

### PR DESCRIPTION
**Pull request by aerickson**
### Before submitting a _pull request_ make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)
### What is the purpose of your _pull request_?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---
### Description of your _pull request_ and other information

I downloaded https://www.vevo.com/watch/lemaitre/Wait/USUV71402190 and noticed that the featured artist was the artist in the title ("LoLo - Wait" vs "Lemaitre ft. LoLo - Wait"). This fixes the issue.

All tests for Vevo are broken on master. This fixes the code to make 4/5 pass.

### CODE REVIEW

**Comment by aerickson** [comment]
@dstftw any tips on getting this landed? 

Has the project thought about only running the tests for changed modules on pull requests? I'm happy to work on it.

Thanks. :)

File: youtube_dl/extractor/vevo.py [hunk 1]
```diff
@@ -242,8 +259,12 @@ def _real_extract(self, url):
 
             timestamp = parse_iso8601(video_info.get('releaseDate'))
             artists = video_info.get('artists')
-            if artists:
-                artist = uploader = artists[0]['name']
+            for curr_artist in artists:
+                if curr_artist['role'] == 'Featured':
```
Comment (dstftw) [inline youtube_dl/extractor/vevo.py]: Missing `role` should not break the extraction.

Comment (aerickson) [inline youtube_dl/extractor/vevo.py]: Fixed. Thanks.

File: youtube_dl/extractor/vevo.py [hunk 2]
```diff
@@ -202,9 +218,11 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
 
         json_url = 'http://api.vevo.com/VideoService/AuthenticateVideo?isrc=%s' % video_id
-        response = self._download_json(
-            json_url, video_id, 'Downloading video info',
-            'Unable to download info', fatal=False) or {}
+        # this request always returns a 500 (non-fatal in normal use, but causes test errors)
+        # response = self._download_json(
+        #     json_url, video_id, 'Downloading video info',
+        #     'Unable to download info', fatal=False) or {}
```
Comment (dstftw) [inline youtube_dl/extractor/vevo.py]: This change is unrelated.

Comment (aerickson) [inline youtube_dl/extractor/vevo.py]: Removed from the diff.

**Comment by aerickson** [comment]
@dstftw Why was this closed? I've made your requested changes.

**Comment by yan12125** [comment]
It's already merged into master: 8b27d83e4e07064898c5ec842e916c84cf7a1826
